c/build.c

```diff
@@ -146,47 +159,105 @@ void prepend_to_data_file()
+    fprintf(data_file, "\nkuliya_schema *__get_node_by_path(const char* path)\n{");
+    fprintf(data_file, "%s\n", clauses);
+    fprintf(data_file, "\treturn NULL;\n");
+    fprintf(data_file, "}\n");
+    fprintf(data_file, "\n#endif\n");
+
+    fclose(data_file);
+}
+
 /**
  * Save Kuliya struct into `data.h` file.
  * @param schema Kuliya schema object.
  * @param slots_length Length of slots if the current schema has terms.
  * @param json_path Path of the `info.json` file.
  * @returns This function do not return anything.
  */
-void save_to_file(const kuliya_schema *schema, const size_t slots_length, const char *json_path)
+void save_to_file(kuliya_schema *schema, const size_t slots_length, const char *json_path)
 {
     FILE *data_file = fopen(TEMP_FILE, "a");
     size_t prefix_length = strlen("../_data/");
     size_t suffix_length = strlen("/info.json");
     char *path_value = (char *)json_path + prefix_length;
     path_value[strlen(path_value) - suffix_length] = '\0';
-    const char *var_name = replace_char(path_value, '/', '_');
+    replace_char(path_value, '/', '_');
 
-    fprintf(data_file, "kuliya_schema %s = {.name = {.ar = \"%s\", .en = \"%s\", .fr = \"%s\"}, .type = \"%s\"",
-            var_name,
+    fprintf(data_file, "kuliya_schema %s = {.name = {.ar = \"%s\", .en = \"%s\", .fr = \"%s\"}, .type = %s, .terms = NULL};\n",
+            path_value,
             schema->name->ar,
             schema->name->en,
             schema->name->fr,
             schema->type);
+
     if (schema->terms != NULL)
     {
-        fprintf(data_file, ", .terms = {.per_year = %d, .slots = {%d", schema->terms->per_year, schema->terms->slots[0]);
-        for (size_t i = 1; i < slots_length; i++)
-        {
-            fprintf(data_file, ", %d", schema->terms->slots[i]);
-        }
-        fprintf(data_file, "}}");
+        size_t varname_length = strlen(path_value);
+        kuliyas_with_terms[kuliya_with_terms_idx].__varname = malloc(varname_length + 1);
+        memcpy(kuliyas_with_terms[kuliya_with_terms_idx].__varname, path_value, varname_length);
+        kuliyas_with_terms[kuliya_with_terms_idx]
+            .__varname[varname_length + 1] = '\0';
+        kuliyas_with_terms[kuliya_with_terms_idx]
+            .__slots_length = slots_length;
+        kuliyas_with_terms[kuliya_with_terms_idx].terms.per_year = schema->terms->per_year;
+        kuliyas_with_terms[kuliya_with_terms_idx].terms.slots = malloc(slots_length * sizeof(int));
+        memcpy(kuliyas_with_terms[kuliya_with_terms_idx].terms.slots, schema->terms->slots, slots_length * sizeof(int));
+        kuliya_with_terms_idx++;
     }
-    fprintf(data_file, "};\n");
+
+    // Add condition clause for path value
+    char clause[100];
+    char path_name[strlen(path_value)];
+    strcpy(path_name, path_value);
+    replace_char(path_name, '_', '/');
+    sprintf(clause, "\n\tif (STR_EQ(\"%s\", path))\n\t\treturn &%s;\n", path_name, path_value);
```

Comment:
shouldn't we use `switch-case` statement instead? maybe the compiler will optimize it better?

Reply:
In C you can't compare strings in switch cases, only integers (enums are other variant of integer values)

Reply:
right, we can then optimize it later via hashmaps